cmeutils/gsd_utils.py

```diff
@@ -133,3 +133,54 @@ def _validate_inputs(gsd_file, snap, gsd_frame):
     elif snap:
         assert isinstance(snap, gsd.hoomd.Snapshot)
     return snap
+
+
+def snap_delete_types(snap, delete_types):
+    """Create a new snapshot with certain particle types deleted.
+
+    Reads in a snapshot and writes the information (excluding delete_types) to
+    a new snapshot. Does not change the original snapshot.
+
+    Information written:
+        - particles (N, types, position, typeid, image)
+        - configuration (box)
+        - bonds (N, group)
+
+    Parameters
+    ----------
+    snap : gsd.hoomd.Snapshot
+        The snapshot to read in
+
+    Returns
+    -------
+    gsd.hoomd.Snapshot
+        The new snapshot with particles deleted.
+    """
+    new_snap = gsd.hoomd.Snapshot()
+    delete_ids = [snap.particles.types.index(i) for i in delete_types]
+    selection = np.where(~np.isin(snap.particles.typeid, delete_ids))[0]
+    new_snap.particles.N = len(selection)
+    new_snap.particles.types = [
+        i for i in snap.particles.types if i not in delete_types
+    ]
+    typeid_map = {
+        i:new_snap.particles.types.index(e)
+        for i, e in enumerate(snap.particles.types)
+        if e in new_snap.particles.types
+    }
+    new_snap.particles.position = snap.particles.position[selection]
+    new_snap.particles.image = snap.particles.image[selection]
+    new_snap.particles.typeid = np.vectorize(typeid_map.get)(
```

Comment:
Maybe this could be a separate PR, but I feel like we might as well write out all of the information of `snap.particles` available attributes? 

```
'angmom', 'body', 'charge', 'diameter', 'image', 'mass', 'moment_inertia', 'orientation', 'position', 'type_shapes', 'typeid', 'types', 'validate', 'velocity'
```

Reply:
Yes, agreed--let's let this be a separate PR though. I ended up not using this function anyway since I use grits c: